### ISSUE TITLE
Good training, bad test results

### ISSUE THREAD

**Issue by Jacob-Stevens-Haas** [qqwweee/keras-yolo3#283]
My training looks very good, but I'm not detecting anything on test.  I have two classes.  Here's my validation error over 60 epochs:  

![val_loss](https://user-images.githubusercontent.com/37048747/48963987-c880ac00-ef53-11e8-80b8-4addeca82150.PNG)

Training appears to be working correctly.  Anything I should be doing differently?  I've heard of reduce LR on plateau, but how do I change it?  Is it in the cfg file?

**Comment by JunjieZhouwust** [comment]
When i train my own dataset, the loss is reduced to about 20. But val_loss is always nan.And the test results is bad too.

**Comment by kilanny** [comment]
you should mention some info. 
Did you check the test samples are similar to ones the training used? (maybe testing on hard nosiy while training on clean, easy data)
How many samples did you train on ? (maybe too few?)

What I can say afiak, overfitting had occured. Maybe training data are not sufficient / bad

**Comment by JunjieZhouwust** [comment]
> you should mention some info.
> Did you check the test samples are similar to ones the training used? (maybe testing on hard nosiy while training on clean, easy data)
> How many samples did you train on ? (maybe too few?)
> 
> What I can say afiak, overfitting had occured. Maybe training data are not sufficient / bad

I have train my my model in the same dataset by darknet, it  make good performance.

**Comment by kilanny** [comment]
@JunjieZhouwust  Which dataset do you mean? COCO?

**Comment by JunjieZhouwust** [comment]
> @JunjieZhouwust Which dataset do you mean? COCO?

My own dataset ,1 classes, about 1000 samples

**Comment by JunjieZhouwust** [comment]
> @JunjieZhouwust Which dataset do you mean? COCO?



> @JunjieZhouwust Which dataset do you mean? COCO?

My own dataset ,1 classes, about 1000 samples

**Comment by Jacob-Stevens-Haas** [comment]
I have 2 classes, several thousand samples.  The training and test data are all 3D models generated from the same random formula, so they should have *exactly* the same noise characteristics, etc.
@ibraheemalkilanny.

**Comment by kilanny** [comment]
@Jacob-Stevens-Haas I don't know what is your model, but what I have seen is training on some objects like COCO dataset (e.g. car or person detection). I tried to train on small objects like text characters using the published weights but is not succeeding, I think YOLO is not suitable for my case and only designed for large enough objects that have many characteristics (like animals, ImageNet dataset for example)

**Comment by jaelim** [comment]
I'm training a model on a private dataset. training loss started out at >11,000 then quickly dropped to >110 after 4-5 epochs. It stuck at 108. But what seems strange to me is that val loss is always `nan`.

Also, I left the default training epoch to 50, but it seems not the right number setting. I've trained my model with darknet, I've done no more than 5 epochs, and already achieved a high performance.

Any tips on why val loss `nan`? I want to test the model, what's the best way? there's no model validation script found here in this repo.

**Comment by ZhiweiDuan** [comment]
@jaelim Have you solved this problem?

val loss is always nan.

**Comment by JJ306** [comment]
> My training looks very good, but I'm not detecting anything on test. I have two classes. Here's my validation error over 60 epochs:
> 
> ![val_loss](https://user-images.githubusercontent.com/37048747/48963987-c880ac00-ef53-11e8-80b8-4addeca82150.PNG)
> 
> Training appears to be working correctly. Anything I should be doing differently? I've heard of reduce LR on plateau, but how do I change it? Is it in the cfg file?

Hi I Have the same problem. I have 6 classes among of them I can only 3 detected, while other can not be detected. I used yolo3.h5 weights. Do you have any idea how can I rid out from this issue?

**Comment by younghuvee** [comment]
> > My training looks very good, but I'm not detecting anything on test. I have two classes. Here's my validation error over 60 epochs:
> > ![val_loss](https://user-images.githubusercontent.com/37048747/48963987-c880ac00-ef53-11e8-80b8-4addeca82150.PNG)
> > Training appears to be working correctly. Anything I should be doing differently? I've heard of reduce LR on plateau, but how do I change it? Is it in the cfg file?
> 
> Hi I Have the same problem. I have 6 classes among of them I can only 3 detected, while other can not be detected. I used yolo3.h5 weights. Do you have any idea how can I rid out from this issue?

did you have check /log/000/ ? New .h5 was saved here, you need test by useing new weights.

**Comment by eain3314** [comment]
损失很高怎么办
